modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/ObjectEntryOpenAPIContributor.java

```diff
@@ -438,34 +469,104 @@ else if (Objects.equals(parameterName, "relatedObjectEntryId")) {
 		return parameters;
 	}
 
+	private ObjectDefinition _getRelatedObjectDefinition(
+		ObjectRelationship objectRelationship) {
+
+		if (_objectDefinition.getObjectDefinitionId() ==
+				objectRelationship.getObjectDefinitionId2()) {
+
+			return _objectDefinitionLocalService.fetchObjectDefinition(
+				objectRelationship.getObjectDefinitionId1());
+		}
+
+		return _objectDefinitionLocalService.fetchObjectDefinition(
+			objectRelationship.getObjectDefinitionId2());
+	}
+
 	private Map<ObjectRelationship, ObjectDefinition>
-		_getRelatedObjectDefinitionsMap() {
+			_getRelatedObjectDefinitionsMap()
+		throws Exception {
 
 		Map<ObjectRelationship, ObjectDefinition> relatedObjectDefinitionsMap =
 			new HashMap<>();
 
 		List<ObjectRelationship> objectRelationships =
-			_objectRelationshipLocalService.getObjectRelationships(
+			_objectRelationshipLocalService.getAllObjectRelationships(
 				_objectDefinition.getObjectDefinitionId());
 
 		for (ObjectRelationship objectRelationship : objectRelationships) {
-			ObjectDefinition objectDefinition =
-				_objectDefinitionLocalService.fetchObjectDefinition(
-					objectRelationship.getObjectDefinitionId2());
-
 			relatedObjectDefinitionsMap.put(
-				objectRelationship, objectDefinition);
+				objectRelationship,
+				_getRelatedObjectDefinition(objectRelationship));
 		}
 
 		return relatedObjectDefinitionsMap;
 	}
 
+	private Schema _getSchema(
+		ObjectRelationship objectRelationship, String schemaName) {
+
+		ObjectSchema schema = new ObjectSchema();
+
+		String description = StringBundler.concat(
+			"Information about the relationship ", objectRelationship.getName(),
+			" can be embedded with \"nestedFields\".");
+
+		schema.setDescription(description);
+
+		schema.set$ref(schemaName);
+
+		if (((objectRelationship.getObjectDefinitionId1() ==
+				_objectDefinition.getObjectDefinitionId()) ||
+			 Objects.equals(
+				 objectRelationship.getType(),
+				 ObjectRelationshipConstants.TYPE_MANY_TO_MANY)) &&
+			(schemaName != null)) {
```

Comment:
With the `schemaName != null` condition, even if the Schema has to be an array, you are setting as a unique element, for example in this relationship:

`University` `oneToMany` `Students`

Request the OpenAPI from Students, then Students has a field called universityStudents with One University, but the Schema of the University contains the same field indicating that is only one element instead an array



![image](https://user-images.githubusercontent.com/32699538/219600843-bc765a5b-0b26-43a6-9c02-69124ff78385.png)

Reply:
Make sense @ccorreagg